src/should-delete-fork.js

```diff
@@ -18,131 +16,84 @@ const branchIsUseful = (github, repo, branch, parentBranches, callback) => {
-			},
+			throw error;
+		}
+	}
 
-			// Grab all branches
-			(repo, callback) => {
-				github.repos.listBranches(
-					{
-						owner: repo.owner.login,
-						repo: repo.name,
-						per_page: 100
-					},
-					(error, branches) => {
-						if (error) {
-							return callback(error);
-						}
+	return false;
+};
 
-						if (branches.length === 100) {
-							// There are too many branches,
-							// dealing with pagination is not supported
-							shouldDeleteCallback(null, false);
-							return;
-						}
+module.exports = async (github, fork) => {
+	// Grab the repository information
+	const {data: repo} = await github.repos.get({
+		owner: fork.owner.login,
+		repo: fork.name
+	});
 
-						callback(null, repo, branches);
-					}
-				);
-			},
+	// Grab all branches
+	const {data: branches} = await github.repos.listBranches({
+		owner: repo.owner.login,
+		repo: repo.name,
+		per_page: 100
+	});
 
-			// Grab all parent repository branches
-			(repo, branches, callback) => {
-				github.repos.listBranches(
-					{
-						owner: repo.parent.owner.login,
-						repo: repo.parent.name,
-						per_page: 100
-					},
-					(error, parentbranches) => {
-						// If parent repository was deleted, need to preserve the fork
-						if (error && error.status == 404) {
-							shouldDeleteCallback(null, false);
-							return;
-						}
+	if (branches.length === 100) {
+		// There are too many branches,
+		// dealing with pagination is not supported
+		return false;
+	}
 
-						callback(error, repo, branches, parentbranches);
-					}
-				);
-			},
+	// Grab all parent repository branches
+	let parentbranches;
+	try {
+		({data: parentbranches} = await github.repos.listBranches({
+			owner: repo.parent.owner.login,
+			repo: repo.parent.name,
+			per_page: 100
+		}));
+	} catch (error) {
+		// If parent repository was deleted, need to preserve the fork
+		if (error && error.status === 404) {
+			return false;
+		}
 
-			// Compare if for each local branch is useless
-			(repo, branches, parentbranches, callback) => {
-				// Quick common case
-				if (branches.length > parentbranches.length) {
-					callback(null, true);
-					return;
-				}
+		throw error;
+	}
 
-				async.some(
-					branches,
-					(branch, someCallback) => {
-						branchIsUseful(
-							github,
-							repo,
-							branch,
-							parentbranches,
-							(error, useful) => {
-								if (error) {
-									return someCallback(error);
-								}
+	// Compare if for each local branch is useless
+	if (branches.length > parentbranches.length) {
+		// Quick common case
+		return false;
+	}
 
-								someCallback(null, useful);
-							}
-						);
-					},
-					callback
-				);
-			}
-		],
-		(error, someBranchesUseful) => {
-			if (error) {
-				shouldDeleteCallback(error);
-			} else {
-				shouldDeleteCallback(null, !someBranchesUseful);
-			}
+	for (const branch of branches) {
+		// eslint-disable-next-line no-await-in-loop
```

Comment:
The rule only matters when it triggers, and if that’s the place where we disable it for one line, it’s better to disable the rule altogether. At this stage it is only an annoyance and causes us to dislike our linting, which is supposed to help.

I’ve reviewed the no-await-in-loop rule and it seems well-intended, but having too many false positives to be useful. Would you agree to disable the rule globally?

Reply:
I think is ok, this is not technically a false positive; we *do* want to stop the loop because we want to process one thing at a time, in both places, to emulate a non-async loop. 

If you prefer disabling it globally that’s also fine though

Reply:
I meant “false positive” as in the goal of the linter is to find mistakes in the code. “Positive” is when the linter reports a warning. “False” is because this is not a mistake in the code, this is correct code. So the linter has had a false positive on this line, and you had to silence the false positive warning.  The true/false positive/negative terminology is always misleading, because it depends on the speaker’s perspective. :-)

Cool, I’ll disable it globally later, then.